pulsar-client/src/main/java/com/yahoo/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -459,13 +479,38 @@ void connectionOpened(final ClientCnx cnx) {
         log.info("[{}][{}] Subscribing to topic on cnx {}", topic, subscription, cnx.ctx().channel());
 
         long requestId = client.newRequestId();
-        cnx.sendRequestWithId(Commands.newSubscribe(topic, subscription, consumerId, requestId, getSubType(),
-                priorityLevel, consumerName), requestId).thenRun(() -> {
+
+        int currentSize;
+        MessageIdImpl startMessageId;
+        synchronized (this) {
+            currentSize = incomingMessages.size();
+            startMessageId = clearReceiverQueue();
```

Comment:
what if we just clear `incomingMessages.clear()`  if subscription has `Durable` mode, it will save some cpu because we don't need `startMessageId` in that case.??